src/python/deepgnn/graph_engine/data/citation.py

```diff
@@ -264,48 +263,34 @@ def _write_json_graph(
         assert node_size == labels.shape[0]
         with open(graph_file, "w") as fout:
             for nid in range(node_size):
-                tmp = self.to_json_node(
+                tmp = self.to_edge_list_node(
                     nid,
                     node_types[nid],
                     flt_feat=list(features[nid].tolist()),
                     label=int(labels[nid]),
                     neighbors=adj[nid].nonzero()[1],
                 )
                 fout.write(tmp)
-                fout.write("\n")
 
     NODE_TYPE_ID = {"train": 0, "val": 1, "test": 2, "other": 3}
 
-    def to_json_node(
+    def to_edge_list_node(
         self,
         node_id: int,
         node_type: str,
         flt_feat: List[float],
         label: int,
         neighbors: np.ndarray,
     ) -> str:
-        """Convert node to json format."""
+        """Convert node to edge_list format."""
         assert type(flt_feat) is list and type(flt_feat[0]) == float
         assert type(label) is int
         ntype = self.NODE_TYPE_ID[node_type]
-        node = {
-            "node_id": node_id,
-            "node_type": ntype,
-            "node_weight": 1.0,
-            "float_feature": {"0": flt_feat, "1": [label]},
-            "uint64_feature": {},
-            "binary_feature": {},
-            "edge": [
-                {
-                    "src_id": int(node_id),
-                    "dst_id": int(nb),
-                    "weight": float(1),
-                    "edge_type": 0,
-                }
-                for nb in neighbors
-            ],
-        }
-        return json.dumps(node)
+        output = ""
+        output += f"{node_id},-1,{ntype},1.0,float32,{len(flt_feat)},{','.join([str(v) for v in flt_feat])},float32,1,{label}\n"
+        for nb in neighbors:
```

Comment:
1. Use `sorted`.
2. Add a unit test that verifies neighbors are processed in the correct order.

Reply:
I think we should add something to the converter / loader to fail if they are not sorted properly. It is not easy to add a test in test_dataset.

Reply:
Please add a test to the converter.